releases/release-6.3.0.md

```diff
@@ -0,0 +1,469 @@
+
+* TiDB supports two more [window functions](/tiflash/tiflash-supported-pushdown-calculations.md) [#5579](https://github.com/pingcap/tiflash/issues/5579) @[SeaRise](https://github.com/SeaRise) **tw：shichun-0415**
+
+    * `LEAD()`
+    * `LAG()`
+
+* Provide lightweight metadata lock to improve the DML success rate during DDL change (experimental) [#37275](https://github.com/pingcap/tidb/issues/37275) @[wjhuang2016](https://github.com/wjhuang2016) **tw: Oreoxmt**
+
+    TiDB uses the online asynchronous schema change algorithm to support changing metadata objects. When a transaction is executed, it obtains the corresponding metadata snapshot at the transaction start. If the metadata is changed during a transaction, to ensure data consistency, TiDB returns an `Information schema is changed` error and the transaction fails to commit. To solve the problem, TiDB v6.3.0 introduces [metadata lock](/metadata-lock.md) into the online DDL algorithm. To avoid DML errors whenever possible, TiDB coordinates the priority of DMLs and DDLs during table metadata change, and makes executing DDLs wait for the DMLs with old metadata to commit.
+
+* Improve the performance of adding indexes and reduce its impact on DML transactions [#35983](https://github.com/pingcap/tidb/issues/35983) @[benjamin2037](https://github.com/benjamin2037) **tw: Oreoxmt**
+
+    To improve the speed of backfilling when creating an index, TiDB v6.3.0 accelerates the `ADD INDEX` and `CREATE INDEX` DDL operations when the [`tidb_ddl_enable_fast_reorg`](/system-variables.md#tidb_ddl_enable_fast_reorg-new-in-v630) system variable is enabled. When the feature is enabled, the performance of adding indexes is about trippled.
+
+### Security
+
+* TiKV supports the SM4 algorithm for encryption at rest [#13041](https://github.com/tikv/tikv/issues/13041) @[jiayang-zheng](https://github.com/jiayang-zheng)
+
+    Add the [SM4 algorithm](/encryption-at-rest.md) for TiKV encryption at rest. When you configure encryption at rest, you can enable the SM4 encryption capacity by setting the value of the `data-encryption-method` configuration to `sm4-ctr`.
+
+* TiDB supports authentication with the SM3 algorithm [#36192](https://github.com/pingcap/tidb/issues/36192) @[CbcWestwolf](https://github.com/CbcWestwolf) **tw：ran-huang**
+
+    TiDB adds an authentication plugin [`tidb_sm3_password`](/security-compatibility-with-mysql.md) based on the SM3 algorithm. When this plugin is enabled, the user password is encrypted and validated using the SM3 algorithm.
+
+* TiDB JDBC supports authentication with the SM3 algorithm [#25](https://github.com/pingcap/mysql-connector-j/issues/25) @[lastincisor](https://github.com/lastincisor) **tw：ran-huang**
+
+    Authenticating the user password needs client-side support. Now because [JDBC supports the SM3 algorithm](/develop/dev-guide-choose-driver-or-orm.md#java-drivers), you can connect to TiDB using SM3 authentication via TiDB-JDBC.
```

Comment:
The linked document says:
"It is strongly recommended to use version 5.1.49, which is the latest version of JDBC 5.1. Since there is an [unresolved bug](https://bugs.mysql.com/bug.php?id=106252) in the current version 8.0.29, which might cause threads to hang when using TiDB. It is recommended that you do not upgrade to version 8.0 until MySQL JDBC 8.0 merges this fix."

1. As https://github.com/pingcap/tidb/issues/32118 is fixed the mentioned bug no longer affects TiDB. So we should now recommend MySQL Connector/J 8.0.x
2. "JDBC" is the API implemented by many different databases, "MySQL Connector/J" is the MySQL specific JDBC implementation. There is no "JDBC 5.1" as mentioned in the text, the last version of the API is 4.3 (see https://en.wikipedia.org/wiki/Java_Database_Connectivity ). The issue here is that both terms seem to be mixed.
3. We should recommend MySQL Connector/J 8.0.30 instead of 8.0.29. Looks like we didn't update the document after 8.0.30 was released.
4. Why do we use `io.github.lastincisor` in the `pom.xml` instead of an official PingCAP repository?
5. I assume even older TiDB versions can use the 8.0 branch of "TiDB MySQL Connector/J"?
6. Where can I find more information about "automatic TiCDC snapshot maintenance"?
7. Are we planning to upgrade the TiDB branch of MySQL Connector/J to 8.0.30 any time soon?

Reply:
1. this version was fixed [the bug ](https://github.com/pingcap/tidb/issues/32118)  [prepare bug fix](https://github.com/pingcap/mysql-connector-j/pull/2) .
3. we bug fix is fork 8.0.29 , next version update  8.0.30 .
4. PingCAP repository not yet applied。 PM and Legal is agree . applied PingCAP repository next version .
5.  "TiDB MySQL Connector/J" compatible 8.0 protocol .

Reply:
So the text on the linked document should be revised?

Reply:
@dveeden please check [this link](https://docs.pingcap.com/tidb/dev/dev-guide-choose-driver-or-orm#java-drivers) (click the *TiDB-JDBC* tab) for the latest JDBC document. Do you think it should be revised?

Reply:
Yes, maybe something like:
"We recommend the use of MySQL Connector/J 8.0.29 or later. For TiDB v6.2 and earlier we recommend you to use the TiDB version of MySQL Connector/J as that includes a fix for an [unresolved bug](https://bugs.mysql.com/bug.php?id=106252) in MySQL Connector/J that affects older versions of TiDB. Using SM3 based authentication is only supported with the TiDB version of MySQL Connector/J"

I don't really like "TiDB MySQL Connector/J". Should we call it "TiDB Connector/J"? I don't think that's really correct either.

Reply:
Thank you for the suggestion. I created another PR to update the TiDB-JDBC doc. We can continue this discussion [here](https://github.com/pingcap/docs/pull/10565).